test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc4Tests.cs

```diff
@@ -47,56 +66,90 @@ public AspNetMvc4TestsCallTargetClassic(IisFixture iisFixture, ITestOutputHelper
+                      + (enableCallTarget ? ".CallSite" : ".CallTarget")
+                      + (classicMode ? ".Classic" : ".Integrated")
+                      + (enableFeatureFlag ? ".NoFF" : ".WithFF")
+                      + (RuntimeInformation.ProcessArchitecture == Architecture.X64 ? ".X64" : ".X86"); // assume that arm is the same
         }
 
+        public static TheoryData<string, int> Data() => new()
+        {
+            { "/Admin", 200 },
+            { "/Admin/Home", 200 },
+            { "/Admin/Home/Index", 200 },
+            { "/", 200 },
+            { "/Home", 200 },
+            { "/Home/Index", 200 },
+            { "/Home/BadRequest", 500 },
+            { "/Home/identifier", 500 },
+            { "/Home/identifier/123", 200 },
+            { "/Home/identifier/BadValue", 500 },
+            { "/Home/OptionalIdentifier", 200 },
+            { "/Home/OptionalIdentifier/123", 200 },
+            { "/Home/OptionalIdentifier/BadValue", 200 },
+            { "/Home/StatusCode?value=201", 201 },
+            { "/Home/StatusCode?value=503", 503 },
+        };
+
         [Theory]
         [Trait("Category", "EndToEnd")]
         [Trait("RunOnWindows", "True")]
         [Trait("LoadFromGAC", "True")]
-        [MemberData(nameof(AspNetMvc4TestData.WithoutFeatureFlag), MemberType = typeof(AspNetMvc4TestData))]
-        public async Task SubmitsTraces(
-            string path,
-            string expectedAspNetResourceName,
-            string expectedResourceName,
-            HttpStatusCode expectedStatusCode,
-            bool isError,
-            string expectedErrorType,
-            string expectedErrorMessage,
-            SerializableDictionary tags)
+        [MemberData(nameof(Data))]
+        public async Task SubmitsTraces(string path, HttpStatusCode statusCode)
         {
-            await AssertWebServerSpan(
-                path,
-                _iisFixture.Agent,
-                _iisFixture.HttpPort,
-                expectedStatusCode,
-                isError,
-                expectedAspNetErrorType: expectedErrorType,
-                expectedAspNetErrorMessage: expectedErrorMessage,
-                expectedErrorType: expectedErrorType,
-                expectedErrorMessage: expectedErrorMessage,
-                "web",
-                "aspnet-mvc.request",
-                expectedAspNetResourceName,
-                expectedResourceName,
-                "1.0.0",
-                tags);
+            var spans = await GetWebServerSpans(path, _iisFixture.Agent, _iisFixture.HttpPort, statusCode);
+
+            var sanitisedPath = VerifyHelper.SanitisePathsForVerify(path);
+
+            var settings = VerifyHelper.GetSpanVerifierSettings(sanitisedPath, statusCode);
+
+            // Overriding the type name here as we have multiple test classes in the file
+            // Ensures that we get nice file nesting in Solution Explorer
+            await Verifier.Verify(spans, settings)
```

Comment:
Can we also set up Verify to place all snapshots in its own directory? I'm thinking something like `/test/snapshots/<ProjectName>/**/verified.txt`